metrics/internal/collectors/ceph-cluster.go

```diff
@@ -115,3 +124,23 @@ func (c *CephClusterCollector) collectMirrorinDaemonCount(cephClusters []*cephv1
 			cephCluster.Namespace)
 	}
 }
+
+func (c *CephClusterCollector) collectLegacyOSDCount(cephClusters []*cephv1.CephCluster, ch chan<- prometheus.Metric) {
+	for _, cephCluster := range cephClusters {
+		cephStorage := cephCluster.Status.CephStorage
+		legacyOSDCount := 0
+		reason := "LVM-based OSDs on a PVC are deprecated, see documentation on replacing OSDs"
+		if cephStorage != nil && cephStorage.DeprecatedOSDs != nil {
+			legacyOSDCount += len(cephStorage.DeprecatedOSDs[reason])
+			ch <- prometheus.MustNewConstMetric(c.LegacyOSD,
+				prometheus.GaugeValue, float64(legacyOSDCount),
+				cephCluster.Name,
+				cephCluster.Namespace)
+			continue
+		}
+		ch <- prometheus.MustNewConstMetric(c.LegacyOSD,
+			prometheus.GaugeValue, float64(legacyOSDCount),
+			cephCluster.Name,
+			cephCluster.Namespace)
+	}
+}
```

Comment:
```suggestion
func (c *CephClusterCollector) collectLegacyOSDCount(cephClusters []*cephv1.CephCluster, ch chan<- prometheus.Metric) {
	for _, cephCluster := range cephClusters {
		cephStorage := cephCluster.Status.CephStorage
		legacyOSDCount := 0
		reason := "LVM-based OSDs on a PVC are deprecated, see documentation on replacing OSDs"
		if cephStorage != nil && cephStorage.DeprecatedOSDs != nil {
			legacyOSDCount = len(cephStorage.DeprecatedOSDs[reason])
		}
		ch <- prometheus.MustNewConstMetric(c.LegacyOSD,
			prometheus.GaugeValue, float64(legacyOSDCount),
			cephCluster.Name,
			cephCluster.Namespace)
	}
}
```
Would this be a valid approach?

Reply:
i think this will work, good catch